src/invoice/signature.rs

```diff
@@ -414,10 +418,32 @@ impl SecretKeyFile {
         Ok(())
     }
 }
+/// This enumerates the select criteria of the key based on the given Label
+pub enum LabelMatch {
```

Comment:
Sorry for another comment here. This is entirely optional, but just wanted your opinion as this API will be used relatively often and I was wondering if this could be cleaner:

```rust
pub enum LabelMatch<'a> {
    /// Key will be selected with an exact match of the given label.
    /// Matches are case sensitive
    FullMatch(&'a str),
    /// Key will be selected with a partial match of the given label,
    /// Matches are case insensitive.
    ///
    /// For example: "pika" will match "Pikachu" and "puff" will match "Jigglypuff"
    PartialMatch(&'a str),
}
```

This would make the enum cheap to construct and no need to copy the `String`. It also would avoid having an `Option<&LabelMatch>` in a function signature.

We can also just merge and come back to this later, but if you do have an opinion, I'd love to hear it!

Reply:
yeah, I had this with lifetime at first but when having it inside `first_matching_key` function in `main.rs` I had trouble with the life time reference as expected. Since these string are small I thought its okay to not use reference. 
But can I give it another try to have `&'a str` like before

Reply:
Nah, no problem. We'll keep this as is and revisit down the line if we want to try again!